src/app/drive/services/worker.service/upload.service.ts

```diff
@@ -0,0 +1,29 @@
+import { TaskStatus } from '../../../tasks/types';
+
+async function waitForContiueUploadSignal(taskId: string): Promise<void> {
+  return new Promise((resolve) => {
+    const checkProgress = async () => {
```

Comment:
Please extract the `checkProgress` function definition from the Promise callback, this code complexity is a headache